common/smoke-test/Initialize-SmokeTests.ps1

```diff
@@ -191,30 +135,136 @@ foreach ($entry in $manifest) {
+       -TestApplicationSecret $using:TestApplicationSecret `
+       -ProvisionerApplicationId $using:TestApplicationId `
+       -ProvisionerApplicationSecret $using:TestApplicationSecret `
+       -TestApplicationOid $using:TestApplicationOid `
+       -TenantId $using:TenantId `
+       -SubscriptionId $using:SubscriptionId `
+       -Location $using:Location `
+       -Environment $using:Environment `
+       -AdditionalParameters $using:AdditionalParameters `
+       -DeleteAfterHours 24 `
+       -Force `
+       -Verbose `
+       -CI:$using:CI
+  }
 }
 
-Write-Verbose "Writing run-manifest.json"
-($runManifest | ConvertTo-Json -AsArray | Set-Content -Path "$repoRoot/common/smoke-test/run-manifest.json" -Force)
+function Deploy-TestResources {
+  Param([Parameter(Mandatory = $true)] $deployManifest)
 
-Write-Verbose "Writing dependencies into Smoke Test package.json"
-$runnerPackageSpec = Get-Content "$repoRoot/common/smoke-test/package.json" | ConvertFrom-Json -AsHashtable
-$runnerPackageSpec.dependencies = $dependencies
-($runnerPackageSpec | ConvertTo-Json | Set-Content "$repoRoot/common/smoke-test/package.json")
+  $deployedServiceDirectories = @{ }
+  $baseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+  $resourceGroupName = "rg-smoke-$baseName"
+  $dependencies = New-Object 'System.Collections.Generic.Dictionary[string,string]'
+  $runManifest = @()
 
-SetEnvironmentVariable -Name "NODE_PATH" -Value "$PSScriptRoot/node_modules"
+  # Use the same resource group name that New-TestResources.ps1 generates
+  Set-EnvironmentVariable -Name 'AZURE_RESOURCEGROUP_NAME' -Value $resourceGroupName
 
-if ($CI) {
-  # If in CI mark the task as successful even if there are warnings so the
-  # pipeline execution status shows up as red or green
-  Write-Host "##vso[task.complete result=Succeeded; ]DONE"
+  $entryDeployJobs = @()
+
+  # Remove any lingering jobs from this session so output is not polluted
+  Get-Job | Stop-Job
+  Get-Job | Remove-Job
+
+  foreach ($entry in $deployManifest) {
+    if (!(Get-ChildItem -Path "$repoRoot/sdk/$($entry.ResourcesDirectory)" -Filter test-resources.json -Recurse)) {
+      Write-Verbose "Skipping $($entry.ResourcesDirectory): could not find test-resources.json"
+      continue
+    }
+
+    if ($deployedServiceDirectories.ContainsKey($entry.ResourcesDirectory) -ne $true) {
+      Write-Verbose "Starting deploy job for $($entry.ResourcesDirectory)"
+      $job = Start-NewTestResourcesJob $entry $baseName $resourceGroupName
+      $entryDeployJobs += $job
+      $deployedServiceDirectories[$entry.ResourcesDirectory] = $true;
+    }
+    else {
+      Write-Verbose "Skipping resource directory deployment (already deployed) for $($entry.ResourcesDirectory)"
+    }
+
+    Update-SamplesForService $entry
+    Update-SampleDependencies $entry $dependencies
+    $runManifest += $entry
+  }
+
+  Write-Verbose "Waiting for all deploy jobs to finish"
+  Get-Job | Wait-Job
```

Comment:
Does this need some sort of timeout?

Reply:
Should this be `$entryDeployJobs | Wait-Job` to scope it just to the jobs you created?

Reply:
I agree with Wes, if there is anywhere in this script we are blocking until something completes, ideally it would have a reasonable timeout (e.g. double the expected runtime) and print a nice error if it doesn't complete in time.  Otherwise, the pipeline will block until it hits the global pipeline timeout (which is often 1-2 hours or more).

Reply:
Ah! Yeah great idea. Forgot I could just pipe that instead.

Reply:
Re: timeout. My thinking was any functional timeouts will happen as via New-TestResources (though I need to check if it has its own limits on backoff when waiting for resources to deploy). At the script level, my thinking was it's better to leave that responsibility to the pipeline configuration, so it's defined in one place and consistent across jobs.

Reply:
Didn't see @mikeharder's comment for some reason. I didn't realize how long the global timeout was. I'll add a local cap in that case, maybe 10-15 minutes or so.